File: Core/DTOs/Responses/AccountResponse.cs
namespace RR.Core.Responses.Account
{
    /// <summary>
    /// Response básico de conta
    /// </summary>
    public class AccountResponse
    {
        public int Id { get; set; }
        public string UserName { get; set; } = string.Empty;
        public string Mail { get; set; } = string.Empty;
        public string? Phone { get; set; }
        public int AccountPermission { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public bool IsActive { get; set; }
    }

    /// <summary>
    /// Response resumido de conta (para listagens)
    /// </summary>
    public class AccountSummaryResponse
    {
        public int Id { get; set; }
        public string UserName { get; set; } = string.Empty;
        public string Mail { get; set; } = string.Empty;
        public int AccountPermission { get; set; }
        public bool IsActive { get; set; }
    }

    /// <summary>
    /// Response de conta criada
    /// </summary>
    public class AccountCreatedResponse
    {
        public int Id { get; set; }
        public string UserName { get; set; } = string.Empty;
        public string Mail { get; set; } = string.Empty;
        public string? Phone { get; set; }
        public int AccountPermission { get; set; }
        public DateTime CreatedAt { get; set; }
    }


}

File: RR.Application/Client/src/app/domain/dto/response/AccountResponse.ts
export interface AccountResponse {
  id: number;
  userName: string;
  mail: string;
  phone?: string;
  accountPermission: number;
  createdAt: string;
  updatedAt: string;
  isActive: boolean;
}

// Interface para resposta resumida de conta (para listagens)
export interface AccountSummaryResponse {
  id: number;
  userName: string;
  mail: string;
  accountPermission: number;
  isActive: boolean;
}

// Interface para resposta de conta criada
export interface AccountCreatedResponse {
  id: number;
  userName: string;
  mail: string;
  phone?: string;
  accountPermission: number;
  createdAt: string; // ISO date string
}

export interface AccountLookup {
  Id: number;
  UserName?: string;
  Mail?: string;
  AccountPermission: number;
}
